arbos/block_processor.go

```diff
@@ -327,6 +327,11 @@ func ProduceBlockAdvanced(
 				return nil, nil, err
 			}
 
+			// Additional transaction validity check
+			if err = extraTxFilter(chainConfig, header, statedb, state, tx, options, sender, l1Info, result); err != nil {
```

Comment:
We don't want this to be a sequencer hook because we want this filter to be available for delayed messages too. You are right about reverting the state changes in case of an error and we can handle that in the current implementation too.

It makes sense to have two extra filter functions one for pre and post execution, I'll add a new function called `extraPreTxFilter`.

Reply:
ahh, I forgot that the hooks can be set to `NoopSequencingHooks`, makes prefect sense to have the checks separetly then